core/src/main/java/org/fao/geonet/kernel/metadata/DefaultStatusActions.java

```diff
@@ -289,80 +353,26 @@ public static String compileMessageWithIndexFields(String message, String uuid,
-     */
-    protected void informOwners(Set<Integer> metadataIds, String changeDate, String changeMessage, String status)
-        throws Exception {
-
-        String translatedStatusName = getTranslatedStatusName(status);
-        // --- get metadata owners (sorted on owner userid)
-        ResourceBundle messages = ResourceBundle.getBundle("org.fao.geonet.api.Messages", new Locale(this.language));
-        String subject = String.format(messages.getString(
-            "status_email_change_title"),
-            siteName, translatedStatusName, replyToDescr, replyTo, changeDate
-        );
-        String mdChanged = buildMetadataChangedMessage(metadataIds);
-
-        Iterable<Metadata> metadata = this.context.getBean(MetadataRepository.class).findAll(metadataIds);
-        List<User> owners = new ArrayList<User>();
-        UserRepository userRepo = this.context.getBean(UserRepository.class);
-
-        for (Metadata md : metadata) {
-            int ownerId = md.getSourceInfo().getOwner();
-            owners.add(userRepo.findOne(ownerId));
-        }
-
-        processList(owners, subject, status, changeDate, changeMessage, mdChanged);
-
-    }
-
-    /**
-     * Process the users and metadata records for emailing notices.
-     *
-     * @param users         The selected set of users
-     * @param subject       Subject to be used for email notices
-     * @param status        The status being set
-     * @param changeDate    Datestamp of status change
-     * @param changeMessage The message indicating why the status has changed
-     */
-    protected void processList(List<User> users, String subject, String status, String changeDate,
-                               String changeMessage, String mdChanged) throws Exception {
-
-        for (User user : users) {
-            sendEmail(user.getEmail(), subject, status, changeDate, changeMessage, mdChanged);
-        }
-    }
 
     /**
      * Send the email message about change of status on a group of metadata records.
      *
      * @param sendTo        The recipient email address
      * @param subject       Subject to be used for email notices
-     * @param status        The status being set on the records
-     * @param changeDate    Datestamp of status change
-     * @param changeMessage The message indicating why the status has changed
+     * @param message       Text of the mail
      */
-    protected void sendEmail(String sendTo, String subject, String status, String changeDate, String changeMessage, String mdChanged) throws Exception {
-        ResourceBundle messages = ResourceBundle.getBundle("org.fao.geonet.api.Messages", new Locale(this.language));
-        String message = String.format(messages.getString(
-            "status_email_change_text"),
-            changeMessage, mdChanged, siteUrl, status, changeDate);
+    protected void sendEmail(String sendTo, String subject, String message) throws Exception {
```

Comment:
This method uses  `MailSender`  that is deprecated. Also what happens in occurs an exception during the mail sending, fails the full request?

Reply:
> This method uses `MailSender` that is deprecated. Also what happens in occurs an exception during the mail sending, fails the full request?

About the failure, it looks like the initial implementation was doing some kind of checks and use emailNotes variable https://github.com/geonetwork/core-geonetwork/blob/master/core/src/main/java/org/fao/geonet/kernel/metadata/DefaultStatusActions.java#L94-L107 Could probably better probably.

Reply:
Done in https://github.com/geonetwork/core-geonetwork/pull/3209/commits/5551c6b7163e14b168899abe02fe2ede58303b11